pulp_smash/tests/rpm/api_v2/test_upload_publish.py

```diff
@@ -332,3 +336,82 @@ def verify_repo_download(self, repo):
             )
         with self.subTest():
             self.assertEqual(self.rpm, response.content)
+
+
+class UploadRetainsVendorData(utils.BaseAPITestCase):
+    """Test whether one can upload, associate and publish RPMs.
+
+    The test procedure is as follows:
+
+    1. Create a repository.
+    2. Upload an RPM to the first repository, and publish it.
+    3. Test that RPM metadata includes all available fields, including vendor.
+    """
+
+    def test_all(self):
+        """Create a pair of RPM repositories."""
+        if selectors.bug_is_untestable(2781, self.cfg.version):
+            raise unittest.SkipTest('https://pulp.plan.io/issues/2781')
+        rpm = utils.http_get(RPM_WITH_VENDOR_URL)
+        client = api.Client(self.cfg, api.json_handler)
+        body = gen_repo()
+        body['distributors'] = [gen_distributor()]
+        repo = client.post(REPOSITORY_PATH, body)
+
+        # Info about repo distributors is needed when publishing.
+        repo = client.get(repo['_href'], params={'details': True})
+        self.addCleanup(client.delete, repo['_href'])
+        self.addCleanup(client.delete, ORPHANS_PATH)
```

Comment:
BaseAPITestCase does this already. Either drop BaseAPITestCase as a parent class, or drop this line of code.